### PR TITLE
feat: #131: grid layout

### LINKED ISSUE #131: Convert card-based layout to grid-based layout
At the moment, all main /trade components are within cards. We should update this to the no-gap grid design from figma. Should be able to be resized small/medium/large.

### PR DESCRIPTION
**Pull request by VanishMax**
Closes #131 

Implemented the grids that look good both on mobile and desktop, created placeholder sections with loaders to fill the empty space. If some of the blocks won't be needed, we can simply remove or disable them later.

https://github.com/user-attachments/assets/297b1668-e4e5-4972-a4cd-1615b4c2e009

### CODE REVIEW

File: src/pages/trade/ui/history.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      <LoadingCell />
+    </div>
+  );
+};
+
+export const History = () => {
+  return (
+    <div className='pt-4 px-4 pb-0 overflow-hidden'>
+      <div className='grid grid-cols-[repeat(10,1fr)] h-auto max-h-[350px] overflow-auto border-b border-b-other-solidStroke'>
+        <div className='grid grid-cols-subgrid sticky top-0 z-10 col-span-10 text-text-secondary border-b border-other-tonalStroke bg-app-main'>
```
Comment (VanishMax) [inline src/pages/trade/ui/history.tsx:24]: learned CSS subgrid system – works amazing for the small tables like ours

File: tailwind.config.ts [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
+import { withPenumbra, tailwindConfig } from '@penumbra-zone/ui/theme';
 
 export default withPenumbra({
   content: ['./src/**/*.{js,ts,jsx,tsx,mdx,css}'],
   theme: {
     extend: {
       colors: {
         ...tailwindConfig.theme.extend.colors,
+        app: {
+          main: '#0d0d0d',
```
Comment (VanishMax) [inline tailwind.config.ts:10]: found that the background of pages in the new design is not a total black like in Minifront

**Comment by JasonMHasperhoven** [comment]
Nice! The app really starts to shape up now, looking great

**Review by JasonMHasperhoven — APPROVED** [review]

**Comment by grod220** [comment]
Given the code diff, it maybe better as a follow up, but a few things I noticed:

The route-book width seems to tight? Maybe padding values missing?

Screenshot 2024-11-19 at 2 12 06 PM

Resizing causes it to disappear.
Screenshot 2024-11-19 at 2 12 30 PM

The figma designs seem to have different layouts for small, medium, large, XL. Think we are missing a few views in there:
Screenshot 2024-11-19 at 2 12 53 PM

File: tailwind.config.ts (hunk 2 above)
Comment (TalDerei) [reply tailwind.config.ts:10]: Screenshot 2024-11-19 at 9 01 24 PM

**Comment by VanishMax** [comment]
@grod220 Thanks for your comment. It made me go through the layout once again fixing the styles for every possible screen type. I'm proud to say that now the grid looks amazing with every resolution tested by me. 

Also answering some questions:
1. The route book has some more work to do, not just the paddings. @vacekj can you take a look at that after the grid is merged?
2. Route depth tab is disabled, we can even remove it for now until implemented. Didn't create the placeholder for this like I did for the history or trade tabs.

File: src/pages/trade/ui/page.tsx [hunk 3]
```diff
@@ -1,55 +1,55 @@
...
+        'grid-cols-[100%] grid-rows-[auto,auto,auto,auto,auto]',
+        // tablet grid (600px)
+        'tablet:grid-cols-[50%,50%] tablet:grid-rows-[auto,auto,auto,auto]',
+        // desktop grid (>900px)
+        'desktop:grid-cols-[1fr,1fr,300px] desktop:grid-rows-[auto,652px,auto,auto]',
+        // large grid (>1200px)
+        'lg:grid-cols-[1fr,320px,320px] lg:grid-rows-[auto,660px,auto]',
+        // extra large grid (>1600px)
+        'xl:grid-cols-[1fr,320px,320px] xl:grid-rows-[auto,660px,auto]',
+      )}
```
Comment (VanishMax) [inline src/pages/trade/ui/page.tsx:26]: 4 iterations of grid layout implementation led me to these styles. Should be easy enough to extend and add new blocks if needed

File: tailwind.config.ts (hunk 2 above)
Comment (VanishMax) [reply tailwind.config.ts:10]: Thanks, fixed it!

**Pull request merged**